### PR TITLE
tools: support config for API Bench and add TSO API

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: ref #6922

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

no code changed about PD server and PD client
- Manual test (add detailed scripts or steps below)

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: tools/pd-api-bench/cases/cases.go [hunk 1]
```diff
@@ -95,37 +95,52 @@ type GRPCCase interface {
...
+
 // HTTPCase is the interface for all HTTP cases.
 type HTTPCase interface {
 	Case
 	Do(context.Context, pdHttp.Client) error
 }
 
-// HTTPCaseMap is the map for all HTTP cases.
-var HTTPCaseMap = map[string]HTTPCase{
+// HTTPCraeteFn is function type to create GRPCCase.
```
Comment (rleungx) [inline tools/pd-api-bench/cases/cases.go:119]:
```suggestion
// HTTPCraeteFn is function type to create HTTPCase.
```

File: tools/pd-api-bench/config/config.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	Burst int64 `toml:"burst" json:"burst"`
+}
+
+// NewConfig return a set of settings.
+func NewConfig(flagSet *flag.FlagSet) *Config {
+	cfg := &Config{}
+	cfg.flagSet = flagSet
+	fs := cfg.flagSet
+	fs.StringVar(&cfg.configFile, "config", "", "config file")
+	fs.StringVar(&cfg.PDAddr, "pd", "http://127.0.0.1:2379", "pd address")
```
Comment (rleungx) [inline tools/pd-api-bench/config/config.go:60]: How about also supporting the status addr?

Comment (CabinfeverB) [reply tools/pd-api-bench/config/config.go:60]: see #7692

**Review by rleungx — APPROVED** [review]

**Pull request merged**